Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -465,14 +465,24 @@ void WebAutomationSession::switchToBrowsingContext(const Inspector::Protocol::Au
     ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!page, WindowNotFound);
 
     bool frameNotFound = false;
-    webFrameIDForHandle(frameHandle, frameNotFound);
+    auto frameID = webFrameIDForHandle(frameHandle, frameNotFound);
     ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(frameNotFound, FrameNotFound);
 
-    // FIXME: Should we also switch to the frame that frameHandle points to?
-    m_client->requestSwitchToPage(*this, *page, [page = Ref { *page }, callback = WTFMove(callback)]() {
+    m_client->requestSwitchToPage(*this, *page, [protectedThis = Ref { *this }, frameID, page = Ref { *page }, callback = WTFMove(callback)]() mutable {
```

Comment:
I don't see any references to `this` in the callbacks, do you really need `protectedThis` here?

Reply:
yeah and also is the `mutable` actually needed?

ditto below